### PR TITLE
Fix rounding issue when converting float64 to int

### PR DESCRIPTION
**Pull request by pschou**

### CODE REVIEW

**Comment by pschou** [comment]
A rounding issue is introduced when converting to and from lab color space.  This commit fixes the rounding issue with minimal impact.

File: oklab.go [hunk 1]
```diff
@@ -33,7 +33,7 @@ var OklchModel = color.ModelFunc(oklchModel)
 func (c Oklab) RGBA() (uint32, uint32, uint32, uint32) {
 	r, g, b := c.SRGB()
 	r, g, b = clampf(r), clampf(g), clampf(b)
-	return uint32(0xffff * r), uint32(0xffff * g), uint32(0xffff * b), 0xffff
+	return (uint32(0x3fffe*r) + 1) >> 2, (uint32(0x3fffe*g) + 1) >> 2, (uint32(0x3fffe*b) + 1) >> 2, 0xffff
```
Comment (alltom) [inline oklab.go]: Could you add a comment that this is like `uint32(0xffff * r)` but rounds to the nearest integer instead of always rounding down?

Comment (pschou) [reply oklab.go]: Yes, this is like what you said, but instead of adding an `if` statement, an addition of an add and bit shift is done instead.

The logic is this:

Instead of operating in the units place, the multiplication by 4 the value of 0xffff ensures we operate on approximately a 1/4 value step.  Instead of doing x4 we could to x2, but this is up for debate.

Take for example, the 4x resolution:

At 4x resolution, every value of 0 becomes 0, 1 becomes 4, 2 becomes 8, and etc.

Imagine a float value of 7.001, multiplied by 4 and turned into an integer, the result is 28. When 1 is added, the result is 29.  With the bit shift back the 29/4 becomes 7 again.

Likewise, when an original value is 2.801, the multiplication and int becomes 11.  Adding 1 bumps the value up to 12 so the bitshift back the 12/4 becomes 3.

In conclusion... By multiplying by 0x3fffe, one ensures that the bounds are set between 0x0 and 0x3fffe, thus adding 1 does not cause any carry bit(s) to roll up to 0x40000.  Bit shifting to the right by two places makes anything 0.75 and less round down, likewise above round-up. If you want to do half-rounding then I am happy to replace this line with this:

```golang
return (uint32(0x1fffe*r) + 1) >> 1, (uint32(0x1fffe*g) + 1) >> 1, (uint32(0x1fffe*b) + 1) >> 1, 0xffff
```

Comment (alltom) [reply oklab.go]: Thanks, that makes a lot of sense! I'll trust your judgement on which variation is appropriate, since they're both within the tolerance of everything I've worked on so far. :) I appreciate the patch.
